test/e2e/reconciler/cache/replication_test.go

```diff
@@ -249,6 +251,42 @@ func replicateShardNegativeScenario(ctx context.Context, t *testing.T, server fr
 	)
 }
 
+// replicateWorkspaceTypeScenario tests if a WorkspaceType is propagated to the cache server.
+// The test exercises creation, modification and removal of the Shard object.
+func replicateWorkspaceTypeScenario(ctx context.Context, t *testing.T, server framework.RunningServer, kcpShardClusterDynamicClient kcpdynamic.ClusterInterface, cacheKcpClusterDynamicClient kcpdynamic.ClusterInterface) {
+	t.Helper()
+	replicateResource(ctx,
+		t,
+		server,
+		kcpShardClusterDynamicClient,
+		cacheKcpClusterDynamicClient,
+		"",
+		"replicate-workspace-type",
+		"WorkspaceType",
+		tenancyv1alpha1.SchemeGroupVersion.WithResource("workspacetypes"),
+		&tenancyv1alpha1.WorkspaceType{ObjectMeta: metav1.ObjectMeta{Name: "replicate-workspace-type"}, Spec: tenancyv1alpha1.WorkspaceTypeSpec{}},
```

Comment:
on the next iteration, we could change the scenarios to only provide a resource and some type information.